ebr.go

```diff
@@ -0,0 +1,263 @@
+			}
+			if isConditions {
+				if err := entityRules.execute(re, rule.Do); err != nil {
+					return fmt.Errorf("errors occured in actions execution: %w", err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (entityRules *EntityRules) isRuleTriggered(entity reflect.Value, when []string) (isTrigger bool, err error) {
+	//Check if at least one trigger is valid using When attribute of the current rule
+	isTrigger = true
+	for _, name := range when {
+		if isTrigger, err = entityRules.executeMethodBool(entity, name); err != nil {
+			return false, fmt.Errorf("%w: %s", err, name)
+		}
+		if isTrigger {
+			break
+		}
+	}
+	return isTrigger, nil
+}
+
+func (entityRules *EntityRules) entityHasConditions(entity reflect.Value, on string) (isConditions bool, err error) {
+	//Check if conditions are valid using If attribute of the current rule
+	if on == "" { //If no conditions are requiered
+		return true, nil
+	}
+	//Formating r.on for latter evaluation expression
+	listParser := []string{"not", "or", "and", "(", ")"}
+	namedConditions := make(map[string]bool)
+	var expression string
+	for _, name := range strings.Split(on, " ") {
+		if !Exists(listParser, name) {
+			if namedConditions[name], err = entityRules.executeMethodBool(entity, name); err != nil {
+				return false, fmt.Errorf("%w: %s", err, name)
+			}
+			expression += " " + name + " == true"
+		} else {
+			expression += " " + name
+		}
+	}
+	expression = strings.TrimLeft(expression, " ")
+	//Evaluating expression with bexpr package
+	eval, err := bexpr.CreateEvaluatorForType(expression, nil, (map[string](bool))(nil))
+	if err != nil {
+		return false, err
+	}
+	isConditions, err = eval.Evaluate(namedConditions)
+	if err != nil {
+		return false, err
+	}
+	return isConditions, nil
+}
+
+func (entityRules *EntityRules) execute(entity reflect.Value, do []string) error {
+	//Execute actions from Do attributes in the current rule
+	for _, name := range do {
+		if _, err := executeMethod(entity, entityRules.Actions, name); err != nil {
+			return fmt.Errorf("%w: %s", err, name)
+		}
+	}
+	return nil
+}
+
+func (entityRules *EntityRules) executeMethodBool(entity reflect.Value, name string) (result bool, err error) {
+	//Evaluating with entity the method while checking if it return a boolean
+	defer func() {
+		if r := recover(); r != nil {
+			result, err = false, r.(error)
+		}
+	}()
+	resultMethod, err := executeMethod(entity, entityRules.Conditions, name)
+	if err != nil {
+		return false, err
+	}
+	result, ok := resultMethod[0].Interface().(bool)
+	if !ok {
+		return false, ErrBool
+	}
+	return result, nil
+}
+
+func executeMethod(entity reflect.Value, methods MethodsExecution, name string) (result []reflect.Value, err error) {
+	//Evaluating with entity the method identified by name and contained in the input methods map
+	defer func() {
+		if r := recover(); r != nil {
+			result, err = nil, r.(error)
+		}
+	}()
+	idx := found(methods, name)
```

Comment:
you can add a method on `MethodsExecution` called `GetByName` =>

`methods.GetByName(name)`, would be better